### ISSUE TITLE
Datapusher cannot push to CKAN using docker-compose on OSX

### ISSUE THREAD

**Issue by spikeheap** [ckan/ckan#3983]
Thanks for the great efforts pushing the Docker setup forward. I've come up against an issue getting Datastore/Datapusher running within docker-compose.

### CKAN Version if known (or site URL)
2.7.2 (with Dockerfile/docker-compose.yml/etc. backported)
Docker for Mac 17.12.0-ce-mac46

### Please describe the expected behaviour

Given the default `.env` template and docker-compose setup, when I run `docker-compose up ckan` and upload a new resource, then datapusher is called and pushes the file into datastore.

### Please describe the actual behaviour

On OSX, with `CKAN_SITE_URL=http://localhost:5000`: CKAN is accessible and works, however when a new file is uploaded the `datapusher` service throws the following error (once per resource).

```
datapusher    | Job "push_to_datastore (trigger: RunTriggerNow, run = True, next run at: None)" raised an exception
datapusher    | Traceback (most recent call last):
datapusher    |   File "/usr/local/lib/python2.7/site-packages/apscheduler/scheduler.py", line 512, in _run_job
datapusher    |     retval = job.func(*job.args, **job.kwargs)
datapusher    |   File "/usr/src/app/datapusher/jobs.py", line 222, in push_to_datastore
datapusher    |     resource = get_resource(resource_id, ckan_url, api_key)
datapusher    |   File "/usr/src/app/datapusher/jobs.py", line 178, in get_resource
datapusher    |     'Authorization': api_key}
datapusher    |   File "/usr/local/lib/python2.7/site-packages/requests/api.py", line 94, in post
datapusher    |     return request('post', url, data=data, json=json, **kwargs)
datapusher    |   File "/usr/local/lib/python2.7/site-packages/requests/api.py", line 49, in request
datapusher    |     return session.request(method=method, url=url, **kwargs)
datapusher    |   File "/usr/local/lib/python2.7/site-packages/requests/sessions.py", line 457, in request
datapusher    |     resp = self.send(prep, **send_kwargs)
datapusher    |   File "/usr/local/lib/python2.7/site-packages/requests/sessions.py", line 569, in send
datapusher    |     r = adapter.send(request, **kwargs)
datapusher    |   File "/usr/local/lib/python2.7/site-packages/requests/adapters.py", line 407, in send
datapusher    |     raise ConnectionError(err, request=request)
datapusher    | ConnectionError: ('Connection aborted.', error(99, 'Cannot assign requested address'))
```

The file is still present in FileStore, and the resource has been created. The Data Explorer pane provides the following error:

```
This resource view is not available at the moment. Click here for more information.

Could not load view: DataProxy returned an error (Data transformation failed. error: An error occured while connecting to the server: DNS lookup failed for URL: http://localhost:5000/dataset/0dca0562-f939-4b4b-aac9-3bcb7559a6a0/resource/d078b9b6-fb4b-4320-b26a-9e3d5735a49a/download/sample.csv)
```

#### Custom hostname set
If we update `.env` with `CKAN_SITE_URL=http://docker.for.mac.localhost:5000`, and upload data using http://docker.for.mac.localhost:5000, datapusher and datastore behave as expected.

Accessing the site through http://localhost:5000 is still _possible_, but it breaks things, for example the data upload widget fails because of CORS, even though the URL is valid:

```
Failed to load http://docker.for.mac.localhost:5000/api/i18n/en: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://localhost:5000' is therefore not allowed access.
```

It looks like this issue will affect Windows as well, as they use `docker.for.win.localhost`, but I haven't validated this.

### What steps can be taken to reproduce the issue?

On a Mac:

```bash
git clone git@github.com:ckan/ckan
cd ckan/contrib/docker
cp .env.template .env
docker-compose up
```

Enable the datastore/datapusher extensions, restart, then try to add an organisation, dataset and CSV resource. The datapusher service throws the error above. You can tail the logs for a single service with `docker-compose logs -f datapusher`.

**Comment by Nimphal** [comment]
Just to add to this, as the above is Mac-specific. 
I am running `ubuntu 16.04 LTS`, `Docker version 17.12.0-ce, build c97c6d6` and `docker-compose version 1.18.0, build 8dd22a9` . I get the same error as detailed above.

**Comment by spikeheap** [comment]
Hi @Nimphal, as this issue was closed by #3984 being merged (and I mistakenly thought it only affected OSX), it might be worth opening a new issue for the Ubuntu version of this problem, and referencing this.
